go-controller/pkg/ovn/egressgw.go

```diff
@@ -343,6 +359,13 @@ func (oc *DefaultNetworkController) deletePodExternalGW(pod *kapi.Pod) (err erro
 	if podRoutingNamespaceAnno == "" {
 		return nil
 	}
+	local, err := oc.isPodInLocalZone(pod)
```

Comment:
we do not want this check here, because if a pod gw was deleted on another node we need to go in and delete all the routes for pods on the local node.